runtime/src/accounts_db.rs

```diff
@@ -1791,17 +1993,30 @@ impl AccountsDB {
             if rvs.is_empty() {
                 storage.set_status(AccountStorageStatus::Full);
 
-                // See if an account overflows the default append vec size.
-                let data_len = (with_meta[infos.len()].1.data.len() + 4096) as u64;
-                if data_len > self.file_size {
-                    self.create_and_insert_store(slot, data_len * 2);
+                // See if an account overflows the append vecs in the slot.
+                let data_len = (with_meta[infos.len()].1.data.len() + STORE_META_OVERHEAD) as u64;
```

Comment:
In the old implementation, this was `(with_meta[infos.len()].1.data.len() + 4096` (I think to account for all the metas stored here: https://github.com/solana-labs/solana/blob/master/runtime/src/append_vec.rs#L420-L425), but `STORE_META_OVERHEAD = 256 < 4096`

Reply:
I added a test to verify the constant value.